### DISCUSSION TITLE
Use of md5 which is an unsecure hash algorithm

### DISCUSSION THREAD

**Discussion by bgaber** [cloud-custodian/cloud-custodian#7434] [General]
### Describe the bug

c7n/resources/securityhub.py uses the md5 hash on lines 460 and 462 and c7n/resources/ssm.py uses md5 on line 553. md5 is now known as a vulnerable hash algorithm. Is it possible to change these three lines of code to use a secure hash?

### What did you expect to happen?

Expect Cloud Custodian to use a secure hash algorithm.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Custodian:   0.9.13
Python:      3.6.8 (default, Nov 16 2020, 16:55:22)
             [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
Platform:    posix.uname_result(sysname='Linux', nodename='SPLAWSSREDKR01', release='3.10.0-1160.24.1.el7.x86_64', version='#1 SMP Thu Apr 8 19:51:47 UTC 2021', machine='x86_64')
Using venv:  True
Docker: False
Installed:

argcomplete==1.12.3
attrs==21.2.0
boto3==1.17.102
botocore==1.20.102
importlib-metadata==4.6.0
jmespath==0.10.0
jsonschema==3.2.0
pyrsistent==0.18.0
python-dateutil==2.8.1
pyyaml==5.4.1
s3transfer==0.4.2
setuptools==39.2.0
six==1.16.0
tabulate==0.8.9
typing-extensions==3.10.0.0
urllib3==1.26.6
zipp==3.4.1
```

**Comment by kapilt** [comment]
we're not really using this for a cryptographic purpose, as such its not clear what the concern would be. our usage is closer to a crc32 / alder checksum. ie its a checksum we store and match alongside two different domain objects to match them based on policy content which is already an elevated construct (aka if you have untrusted users authoring policies, you have bigger issues). this is mostly an artifact of static analyzers wanting to flag md5 afaics due it being being broken for cryptographic scenarios (hashed passwords, collision vulnerabilities ). if you have a specific concern and want to discuss further please send an email to the security team per https://github.com/cloud-custodian/cloud-custodian#security
